### PR TITLE
Rename stepdef names, fix job status and stop() method

### PR DESCRIPTION
**Pull request by srinathgit**
Rename stepdef names, fix job status and stop() method

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/hub-internal-artifacts/steps/ingestion/marklogic/default-ingestion.step.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "language" : "zxx",
   "name" : "default-ingestion",
-  "type" : "INGEST",
+  "type" : "INGESTION",
   "version" : 1,
   "modulePath": "/data-hub/5/builtins/steps/ingest/default/main.sjs",
```
Comment (aebadirad) [inline marklogic-data-hub/src/main/resources/hub-internal-artifacts/steps/ingestion/marklogic/default-ingestion.step.json]: steps/ingestion/default

File: marklogic-data-hub/src/main/resources/hub-internal-artifacts/steps/mastering/marklogic/default-mastering.step.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "language" : "zxx",
   "name" : "default-mastering",
-  "type" : "MASTER",
+  "type" : "MASTERING",
   "version" : 1,
   "modulePath" : "/data-hub/5/builtins/steps/master/default/main.sjs",
```
Comment (aebadirad) [inline marklogic-data-hub/src/main/resources/hub-internal-artifacts/steps/mastering/marklogic/default-mastering.step.json]: steps/mastering/default

**Comment by aebadirad** [comment]
@akshaysonvane pr covers the concerns i had

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs [hunk 3]
```diff
@@ -31,7 +31,7 @@ class Step {
       this.hubUtils = datahub.hubUtils;
       this.performance = datahub.performance;
     }
-    this.stepTypes = ['ingest', 'mapping', 'custom'];
+    this.stepTypes = ['ingestion', 'mapping', 'custom'];
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs]: @aebadirad ,
Should this include 'mastering' as well ?

**Comment by srinathgit** [comment]
@clockworked247 ,
Can you please check if the changes to prov.sjs are correct ?

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs (hunk 3 above)
Comment (aebadirad) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs]: Yep, it should, good catch

**Comment by aebadirad** [comment]
We'll need to add changes to the prov anyways for 'step name' vs. 'step type', so the actual name can be captured too.

We'll add that in a different PR

**Pull request merged**